Jetchat/app/src/main/java/com/example/compose/jetchat/conversation/Conversation.kt

```diff
@@ -224,22 +230,40 @@ fun Messages(
                     isLastMessageByAuthor = isLastMessageByAuthor
                 )
             }
+            // Add some space for the top app bar
+            item {
+                // TODO: Get height from somewhere.
+                Spacer(
+                    modifier = Modifier
+                        .height(90.dp)
+                        .statusBarsPadding()
+                )
+            }
         }
         // Jump to bottom button shows up when user scrolls past a threshold.
         // Convert to pixels:
         val jumpThreshold = with(LocalDensity.current) {
             JumpToBottomThreshold.toPx()
         }
 
-        // Apply the threshold:
-        val jumpToBottomButtonEnabled = scrollState.value > jumpThreshold
+        // Show the button if the first visible item is not the first one or if the offset is
+        // greater than the threshold.
+        val jumpToBottomButtonEnabled = (
+            scrollState.firstVisibleItemIndex != 0 ||
```

Comment:
Please use derivedStateOf to have less recompositions on scroll as described here: https://developer.android.com/jetpack/compose/lists#react-to-scroll-position

Reply:
Ah thanks, first time I use it!